src/Symfony/Component/Routing/Attribute/AsRouteExpressionConditionVariable.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class AsRouteExpressionConditionVariable
+{
+    public function __construct(
+        public string $name,
+        public int $priority = 0,
```

Comment:
what's the usecase for priority?

Reply:
This gives the control about name overriding (different services with the same name).

Reply:
@chapterjason, regarding your comment (https://github.com/symfony/symfony/pull/44405#issuecomment-984847379)

I was thinking to handle the name override via priority

Reply:
@renanbr You mean to override the service?